### PR TITLE
Upgrade sample projects

### LINKED ISSUE #29: Upgrade the sample projects
The example projects are way outdated, using a version of Angular that doesn't exist anymore (_angular2_ package, replaced by _angular_). I did some small tests and it is possible to re-implement the basic examples (so far, I made the _pwa-default_ project work).

Addressing the issue to link with an upcoming pull request, that probably will resolve #22.

### PR DESCRIPTION
**Pull request by muniz95**
For each old project, I created a new, fresh project using _stagehand_ and made the necessary fixes in the code, as some angular APIs changed. Due the fact I use Windows and Linux alternately, I created two versions of each project, to make the examples work in both OSes, as well as a README in the examples folder.

Resolves #29

### CODE REVIEW

**Comment by isoos** [comment]
Wow, thanks, that's a lot, but I'll try to review it soon.

**Comment by robertmuth** [comment]
Just out of curiosity do the examples work with both ddc and dart2js for you?

**Comment by muniz95** [comment]
@robertmuth since I am using webdev to run the newer projects, the default behavior is to use ddc. So yes, this little guy is working for me. I can test the behavior with dart2js in the next days.

**Comment by robertmuth** [comment]
@muniz95 were you able to get it to work with dart2js?

**Comment by muniz95** [comment]
@robertmuth yes, I tested it using webdev with the --release flag (builds the application using dart2js) and saw no issues.

**Comment by isoos** [comment]
Thanks for the patience, I'll likely finish the review before the end of this week.

File: examples/general/custom_routes/lib/pwa/offline_urls.g.dart [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
   './',
+  './.build.manifest',
+  './.packages',
+  './favicon.png',
   './main.dart.js',
-  './main.dart.js_1.part.js',
   './manifest.json',
-  './packages/browser/dart.js',
-  './packages/browser/interop.js',
+  './packages/\$sdk/_internal/ddc_sdk.sum',
```
Comment (isoos) [inline examples/general/custom_routes/lib/pwa/offline_urls.g.dart:9]: Do you happen to know the content of these `.sum` files?

File: examples/general/custom_routes/lib/pwa/offline_urls.g.dart [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+  './packages/\$sdk/_internal/strong.sum',
+  './packages/\$sdk/dev_compiler/amd/dart_sdk.js',
```
Comment (isoos) [inline examples/general/custom_routes/lib/pwa/offline_urls.g.dart:11]: I think these are not present in a prod release (`dart2js`). Although it is unlikely that these will hurt much (won't take up disk space), we should probably include only an example that has the final files cached offline.

File: examples/general/custom_routes/lib/pwa/offline_urls.g.dart [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+  './packages/angular_components/material_select/material_dropdown_select.md',
+  './packages/angular_components/src/material_datepicker/closedhand.cur',
+  './packages/angular_components/src/material_datepicker/openhand.cur',
+  './packages/build_runner/src/server/build_updates_client/hot_reload_client.dart.js',
+  './packages/build_runner/src/server/build_updates_client/live_reload_client.js',
+  './packages/build_runner/src/server/graph_viz.html',
+  './packages/build_runner/src/server/graph_viz.js',
+  './packages/build_runner/src/server/graph_viz_main.dart.js',
+  './packages/build_web_compilers/src/dev_compiler_stack_trace/stack_trace_mapper.dart.js',
+  './packages/front_end/src/fasta/TESTING.md',
```
Comment (isoos) [inline examples/general/custom_routes/lib/pwa/offline_urls.g.dart:36]: Pretty sure these should be filtered out.

File: examples/general/custom_routes/lib/pwa/offline_urls.g.dart [hunk 4]
```diff
@@ -224,10 +267,23 @@ final List<String> offlineUrls = [
...
   './packages/intl/src/data/dates/symbols/zh_TW.json',
   './packages/intl/src/data/dates/symbols/zu.json',
-  './packages/pwa_example/app/app_component.html',
-  './packages/pwa_example/app/deferred_component.html',
+  './packages/kernel/binary/readme.md',
+  './packages/kernel/target/readme.md',
+  './packages/kernel/text/readme.md',
+  './packages/node_preamble/preamble.js',
+  './packages/node_preamble/preamble.min.js',
+  './packages/pedantic/analysis_options.1.0.0.yaml',
```
Comment (isoos) [inline examples/general/custom_routes/lib/pwa/offline_urls.g.dart:275]: These too.

**Comment by isoos** [comment]
@muniz95 wdyt about the list of offline urls?

**Comment by muniz95** [comment]
Well, as they are not related to the projects migration _per se_, I believe they can be treated in a separated issue.

**Comment by isoos** [comment]
> Well, as they are not related to the projects migration per se, I believe they can be treated in a separated issue.

Okay, let's merge this PR, and if somebody has time, hopefully they will fix those issues. Thanks for the PR anyways :)

**Pull request merged**